### PR TITLE
Regress Olympia on MacOS

### PR DESCRIPTION
**Pull request by kathlenemagnus**
Expanding the Olympia regression to build and run on MacOS-12. I'm not sure if a separate Conda env file is necessary. Testing it with `conda/environment.yml` to start.

### CODE REVIEW

**Comment by kathlenemagnus** [comment]
So it looks like there are a lot of libraries in `conda/environment.yml` that are not available with the macos-12 runner image. I'll need to create a separate Conda env yml and figure out library versions are available: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md

File: .github/workflows/conda/macos_env.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: riscv_perf_model
```
Comment (kathlenemagnus) [inline .github/workflows/conda/macos_env.yml:1]: I copied `conda/environment_from_history.yml` and then made a couple of adjustments to create this file.

**Comment by kathlenemagnus** [comment]
Current failure:
```
ld: library not found for -lyaml-cpp
```

File: .github/workflows/conda/macos_env.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - doxygen>=1.8.20
+  - cmake>=3.25.2
+  - hdf5>=1.10.6
+  - sqlite>=3.40.0
+  - boost>=1.76.0
+  - boost-cpp>=1.76.0
+  - xz>=5.2.6
+  - yaml-cpp>=0.7.0
+  - rapidjson>=1.1.0
+  - cppcheck>=2.7.5
```
Comment (kathlenemagnus) [inline .github/workflows/conda/macos_env.yml:14]:
```suggestion
  - python=3.10.*                                                                                   
  - python_abi=3.*                                                                                  
  - python-dateutil=2.8.*                                                                           
  - pyyaml=6.*                                                                                      
  - rapidjson=1.1.*                                                                                 
  - readline=8.1.*                                                                                  
  - requests=2.28.*                                                                                 
  - rhash=1.4.*                                                                                     
  - rsync=3.2.*                                                                                     
  - setuptools=67.4.*                                                                               
  - sigtool=0.1.*                                                                                   
  - six=1.16.*                                                                                      
  - sqlite=3.40.*                                                                                   
  - stack_data=0.6.*                                                                                
  - sysroot_linux-64=2.*                                                                            
  - tapi=1100.0.*                                                                                   
  - texlive-core                                                                                    
  - tk=8.6.*                                                                                        
  - traitlets=5.9.*                                                                                 
  - tzdata                                                                                          
  - unicodedata2=15.0.*                                                                             
  - urllib3=1.26.*                                                                                  
  - wcwidth=0.2.*                                                                                   
  - wheel=0.38.*                                                                                    
  - wxpython=4.2.*                                                                                  
  - xorg-kbproto=1.0.*                                                                              
  - xorg-libice=1.0.*                                                                               
  - xorg-libsm=1.2.*                                                                                
  - xorg-libx11=1.7.*                                                                               
  - xorg-libxau=1.0.*                                                                               
  - xorg-libxdmcp=1.1.*                                                                             
  - xorg-libxext=1.3.*                                                                              
  - xorg-libxrender=0.9.*                                                                           
  - xorg-renderproto=0.11.*                                                                         
  - xorg-xextproto=7.3.*                                                                            
  - xorg-xproto=7.0.*                                                                               
  - xxhash=0.8.*                                                                                    
  - xz=5.2.*                                                                                        
  - yaml=0.2.*                                                                                      
  - yaml-cpp=0.7.*                                                                                  
  - zlib=1.2.*                                                                                      
  - zstd=1.5.*
```

**Comment by phattso** [comment]
Running into the same missing `yaml-cpp` library issue at link time

**Comment by phattso** [comment]
Success!
Looks like limiting yaml-cpp to v0.7 instead of the latest v0.8 seems to have done the trick

**Review by klingaard — APPROVED** [review]

**Pull request merged**